coaster/sqlalchemy/roles.py

```diff
@@ -293,13 +293,28 @@ class RoleAccessProxy(collections.Mapping):
 
     :param obj: The object that should be wrapped with the proxy
     :param roles: A set of roles to determine what attributes are accessible
+    :param actor: The actor this proxy has been constructed for
+    :param anchors: The anchors this proxy has been constructed with
+    :param profiles: Data profiles to limit attribute enumeration to
+
+    The `actor` and `anchors` parameters are not used by the proxy, but are used to
+    construct proxies for objects accessed via relationships.
     """
 
-    def __init__(self, obj, roles, actor, anchors):
+    def __init__(self, obj, roles, actor, anchors, data_profiles):
         object.__setattr__(self, '_obj', obj)
         object.__setattr__(self, 'current_roles', InspectableSet(roles))
         object.__setattr__(self, '_actor', actor)
         object.__setattr__(self, '_anchors', anchors)
+        if data_profiles is None:
+            data_profile_attrs = None
+            object.__setattr__(self, '_data_profiles', None)
+        else:
+            # This will raise an IndexError if profiles is empty, an expected outcome
+            # signalling that data profiles have been misconfigured.
+            data_profile_attrs = obj.__data_profiles__[data_profiles[0]]
```

Comment:
I don't think we should do this. It will force definition of data profiles on all models just to use one. Instead, this piece of code should set `data_profile_attrs = set()` so that the object will appear to be empty under enumeration.